evap/staff/views.py

```diff
@@ -1207,6 +1207,15 @@ def notify_reward_points(grantings, **_kwargs):
 
         return redirect("staff:semester_view", semester.id)
 
+    questionnaires_with_answers_per_contributor = {}
+    for contribution in evaluation.contributions.all():
+        questionnaires = Questionnaire.objects.filter(
```

Comment:
Hmm, this looks easy at first glance, but after trying for quite a while, I feel like this is another one of those queries where django just doesn't quite provide the flexibility -- all my attempts to direct select questionnaires or contributors failed.

Essentially, we want (which isn't really that complicated)
```sql
SELECT UserProfile, Questionnaire
FROM (pairwise inner joins) UserProfile, Contribution, Questionnaire, Question, Answer
WHERE Answer.contribution == Contribution.id AND Contribution.evaluation = INPUT;
```
The best code I could come up with is starting at answer, since we can follow foreign-keys from here to questionnaire and to contribution. I noticed this code already passes the test:
```python3
    answers = [
        *TextAnswer.objects.select_related("question__questionnaire", "contribution__contributor"),
        *RatingAnswerCounter.objects.select_related("question__questionnaire", "contribution__contributor"),
    ]

    questionnaires_with_answers_per_contributor = defaultdict(list)
    for answer in answers:
        questionnaires_with_answers_per_contributor[answer.contribution.contributor].append(
            answer.question.questionnaire
        )
```

Even though it doesn't filter for the correct evaluation, because in the test, there are no answers for other evaluations. Adding this to the view test's setUpTestData solves that:
```python3
        baker.make(
            TextAnswer, question__questionnaire=cls.contributor_questionnaire, contribution__contributor=cls.editor
        )
```

fixing the filtering and adding some syntactic sugar leaves us at:
```python3
    contributor_questionnaire_pairs = [
        (answer.contribution.contributor, answer.question.questionnaire)
        for answer_cls in [TextAnswer, RatingAnswerCounter]
        for answer in answer_cls.objects.filter(contribution__evaluation=evaluation).select_related(
            "question__questionnaire", "contribution__contributor"
        )
    ]

    questionnaires_with_answers_per_contributor = defaultdict(list)
    for (contributor, questionnaire) in contributor_questionnaire_pairs:
        questionnaires_with_answers_per_contributor[contributor].append(questionnaire)
```

it's 10 lines instead of 8, and a bit harder to read. What do you think, is it worth it?

@niklasmohrin is there a one-liner to aggreate a list of (key, value) pairs into a dict[str, list[value-type]]?